Solutions/Marain.TenantManagement.Abstractions/Marain/TenantManagement/TenantManagementServiceExtensions.cs

```diff
@@ -224,5 +225,36 @@ public static class TenantManagementServiceExtensions
 
             return await tenantManagementService.GetServiceEnrollmentConfigurationRequirementsAsync(serviceTenant).ConfigureAwait(false);
         }
+
+        /// <summary>
+        /// Adds arbitrary configuration for a tenant.
+        /// </summary>
+        /// <param name="tenantManagementService">The <see cref="ITenantManagementService"/>.</param>
+        /// <param name="tenantId">The ID of the tenant to enroll.</param>
+        /// <param name="configurationItems">Configuration to add.</param>
+        /// <returns>A task which completes when the configuration has been added.</returns>
+        public static async Task AddConfigurationAsync(
+            this ITenantManagementService tenantManagementService,
+            string tenantId,
+            ConfigurationItem[] configurationItems)
+        {
+            if (string.IsNullOrWhiteSpace(tenantId))
+            {
+                throw new ArgumentException(nameof(tenantId));
+            }
+
+            if (configurationItems == null)
+            {
+                throw new ArgumentNullException(nameof(configurationItems));
+            }
+
+            ITenant tenant = await tenantManagementService.GetTenantOfTypeAsync(
+                tenantId,
+                MarainTenantType.Client,
+                MarainTenantType.Delegated,
+                MarainTenantType.Service).ConfigureAwait(false);
```

Comment:
As above, I don't think it's a great idea to allow adding arbitrary config to `MarainTenantType.Delegated` tenants, and not sure why you'd be adding to `MarainTenantType.Service` tenants.

Reply:
Limited to client tenant